src/main/323C.c

```diff
@@ -471,7 +471,65 @@ void load_vram_rect_ptrs(void)
     vram_rect_ptr = &vram_rect_ptrs[0];
 }
 
-INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_80015ECC);
+extern s32 D_8016DEA8;
+extern s32 D_8016EEA8;
+
+void decompress_player_gfx(struct PlayerObj* arg0, s16 arg1, s16 arg2)
+{
+    u8* var_s1;
+    s32 temp_v0;
+    s16 var_s2;
+    s32* new_var;
+    u8* temp_a2;
+    s32 temp_v1_3;
+    s32 var_v0;
+    s32 var_v0_2;
+    struct RectPtrPair* var_a1;
+    u32 temp_a1;
+    s16 var_s0;
+    var_s2 = arg2;
+    if (arg0->prev_anim != arg0->cur_anim) {
+        temp_a2 = arg0->unk38;
+        new_var = arg0->unk38;
+        arg0->prev_anim = arg0->cur_anim;
+        temp_a1 = new_var[arg0->cur_anim];
+        var_s0 = temp_a1 >> 0x14;
+        if (arg0->unk49 == 3) {
+            var_s1 = &D_8016DEA8;
+        } else {
+            var_s1 = (arg0->unk49 << 0xa) + (&D_8016EEA8);
+        }
+        decompress_gfx(&temp_a2[temp_a1 & 0xFFFFF], var_s1, temp_a2);
```

Comment:
Seems like this third argument is likely fake

Reply:
Seems so, adjusted decompress_gfx